opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -165,20 +165,10 @@ def create(
         """
         if not attributes:
             attributes = {}
-        resource = _DEFAULT_RESOURCE.merge(
-            OTELResourceDetector().detect()
-        ).merge(Resource(attributes, schema_url))
-        if not resource.attributes.get(SERVICE_NAME, None):
-            default_service_name = "unknown_service"
-            process_executable_name = resource.attributes.get(
-                PROCESS_EXECUTABLE_NAME, None
-            )
-            if process_executable_name:
-                default_service_name += ":" + process_executable_name
-            resource = resource.merge(
-                Resource({SERVICE_NAME: default_service_name}, schema_url)
-            )
-        return resource
+
+        return get_aggregated_resources(
```

Comment:
I still think simply adding `_DEFAULT_RESOURCE ` to `OTELResourceDetector ` is a more elegant solution. It's also kind of weird that we are forcing `create` to have the same codepath as `get_aggregated_resources`, since the latter takes in `detectors` but the former has nothing to do with detectors.

Reply:
### Reply

If we do that though, then the `service.name` won't default to `unknown_service:test` as is required [in the specs](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service). If users simply use `get_aggregated_resources` alone, they would have to do 

```python
my_attributes_resource = Resource.create(my_attributes)
my_aggregated_resource = get_aggregated_resources([MyResourceDetector])
my_resource = my_aggregated_resource.merge(my_attributes_resource)
```

for example, in which case `OTELResourceDetector`will get called twice by `get_aggregated_resources` and `Resource.create()`.

We cannot use `initial_resource=` as I explained [this comment on the PR](https://github.com/open-telemetry/opentelemetry-python/pull/2013#discussion_r680263079) because subsequent detectors will **overwrite** attributes from the original `my_attributes`.

> It's also kind of weird that we are forcing create to have the same codepath as get_aggregated_resources, since the latter takes in detectors but the former has nothing to do with detectors.

I think this is okay... `Resource.create()` is _already_ using `OTELResourceDetector` right? So it's not entirely independent of "detectors" as it is right now.

https://github.com/open-telemetry/opentelemetry-python/blob/65670cfcf4c2ce23fb9b655fa5b7ad0f50a18f4e/opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py#L166-L171

I think having it return `get_aggregated_resources([])` just rewords what it is _already doing_ in a more clear manner. Except now we move the defaults users should want (`opentelemetry-python business metrics`) into `get_aggregated_resources` so that they only have to use **one** function, not 2. (`Resource.create()` AND `get_aggregated_resources`).

### Action Items

Looking at it more, I think `get_aggregated_resources` and `Resource.create()` have overlapping responsibilities. Because they both use `OTELResourceDetectors`, but only the former should need it.

I think the right answer is to deprecate the `Resource.create(my_attributes)` method, and users should only ever use 
* `Resource()` (with no defaults) or 
* `get_aggregated_resources([])` which would do the same as `Resource.create()` (defaults + `service.name`) right now or
* `get_aggregated_resources([detector1, detector2])` which starts with (defaults + `service.name`), and adds detectors afterwards
* (OPTIONAL) `get_aggregated_resources([detector1, detector2], final_attributes=my_attributes)` which starts with (defaults + `service.name`), adds detectors afterwards, and finally adds my attributes with the highest priority

Reply:
@NathanielRN 

We shouldn't change anything about `Resource.create()`, this is what we recommend to users to do to create a `Resource`. 
I think your change for `get_aggregated_resource()` is actually fine, but instead of having the logic in that, just leave it in `Resource.create()` and call `create()` from within `get_aggregated_resource()`.

Reply:
Thanks @lzchen ! That makes a lot of sense 🙂 .

I updated my PR to do what you said and the tests passed locally so I'll go ahead with that as the solution!

Reply:
Just adding onto that, `get_aggregated_resources()` is only a convenience to run multiple **detectors** in parallel and merge them in order. `OTelResourceDetector` is also just an implementation detail of how we pull in from the envvars. It is exposed publicly which was maybe a mistake, but should be considered an implementation detail of `Resource.create()`.

Reply:
> Just adding onto that, get_aggregated_resources() is only a convenience to run multiple detectors in parallel and merge them in order. 

Hm, so the way you read the method name `get_aggregated_resources(detectors)`, would it surprise you for it to try to set `_DEFAULT_RESOURCE` or `service.name` in the attributes? Because that is what this PR is adding to it.

I can see it going either way as to whether they should be added from a developer's POV.

But as a user, I think I am more likely to forget that I need to do 

```
Resource.create().merge(get_aggregated_resources(detectors)
```

and would just do

```
get_aggregated_resources(detectors)
```

which would make me miss out on attributes set by `Resource.create()`. Then, as a user, I would want `get_aggregated_resources(detectors)` to just handle that for me and set the necessary defaults.

Reply:
> Hm, so the way you read the method name `get_aggregated_resources(detectors)`, would it surprise you for it to try to set `_DEFAULT_RESOURCE` or `service.name` in the attributes? Because that is what this PR is adding to it.

No, it doesn't surprise me. I agree this PR fixes unexpected behavior. I'm just agreeing with Leighton that `get_aggregated_resources([])` should NOT be a pattern we encourage.

Reply:
Sounds good, I didn't realize that [Resource.create()](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#create) was a spec defined function, so it makes sense that `get_aggregated_resources` should call _it_ and not the other way around 🙂